sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -114,7 +113,9 @@ public void onFailure(Throwable e) {
             return Futures.immediateFailedFuture(new AlterTableAliasException(table.ident().fqn()));
         }
 
-        List<ListenableFuture<Long>> results = new ArrayList<>(3);
+        final SettableFuture<Long> result = SettableFuture.create();
+        final AtomicInteger operations = new AtomicInteger(2);
```

Comment:
It avoids having to adjust the number of operations and also doesn't require `if (operations.decrementAndGet() == 0) {` logic at various places.

Imho it is easier to extend, but maybe I'm missing something